gnucash/gnome-utils/gnc-main-window.cpp

```diff
@@ -1885,36 +1901,36 @@ gnc_main_window_update_radio_button (GncMainWindow *window)
     GtkAction *action, *first_action;
     GSList *action_list;
     gchar *action_name;
-    gint index;
+    gsize index;
 
     ENTER("window %p", window);
 
     /* Show the new entry in all windows. */
     index = g_list_index(active_windows, window);
     if (index >= n_radio_entries)
     {
-        LEAVE("window %d, only %d actions", index, n_radio_entries);
+        LEAVE("window  %" G_GSIZE_FORMAT ", only %" G_GSIZE_FORMAT " actions", index, n_radio_entries);
         return;
     }
 
     priv = GNC_MAIN_WINDOW_GET_PRIVATE(window);
-    action_name = g_strdup_printf("Window%dAction", index);
+    action_name = g_strdup_printf("Window%" G_GSIZE_FORMAT "Action", index);
     action = gtk_action_group_get_action(priv->action_group, action_name);
 
     /* Block the signal so as not to affect window ordering (top to
      * bottom) on the screen */
     action_list = gtk_radio_action_get_group(GTK_RADIO_ACTION(action));
     if (action_list)
     {
-        first_action = g_slist_last(action_list)->data;
+        first_action = static_cast<GtkAction*>(g_slist_last(action_list)->data);
         g_signal_handlers_block_by_func(G_OBJECT(first_action),
-                                        G_CALLBACK(gnc_main_window_cmd_window_raise),
+                                        (void*)gnc_main_window_cmd_window_raise,
```

Comment:
Perhaps use (gpointer) here for consistency as you also use that further down in `gnc_main_window_class_init` ?

Reply:
I see there's also `(GFunc)`, but I don't know if it would apply here without looking at the function signature of `gnc_main_window_cmd_window_raise`. (I'm doing this review online).

Reply:
I made it `gpointer` even though I hate that GNOMEism.